File: 1erdepartamental/python/3.calculadora/core/calculadora.py
# Calculadora básica con ABC

from abc import ABC, abstractmethod


class Calculadora(ABC):
    def __init__(self):
        self.__resultado = 0

    @property
    def resultado(self):
        return self.__resultado

    @resultado.setter
    def resultado(self, valor):
        self.__resultado = valor

    def sumar(self, a, b):
        self.__resultado = a + b
        return self.__resultado

    def restar(self, a, b):
        self.__resultado = a - b
        return self.__resultado

    def multiplicar(self, a, b):
        self.__resultado = a * b
        return self.__resultado

    def dividir(self, a, b):
        if b == 0:
            raise ValueError("División por cero")
        self.__resultado = a / b
        return self.__resultado

File: 1erdepartamental/c++/3.calculadora/core/Calculadora.cpp
#include "Calculadora.h"
#include <stdexcept>

Calculadora::Calculadora() : __resultado(0) {}

double Calculadora::getResultado() const {
    return __resultado;
}

double Calculadora::sumar(double a, double b) {
    __resultado = a + b;
    return __resultado;
}

double Calculadora::restar(double a, double b) {
    __resultado = a - b;
    return __resultado;
}

double Calculadora::multiplicar(double a, double b) {
    __resultado = a * b;
    return __resultado;
}

double Calculadora::dividir(double a, double b) {
    if (b == 0) {
        throw std::runtime_error("Error: división por cero");
    }
    __resultado = a / b;
    return __resultado;
}
